### DISCUSSION TITLE
Responses during pre_target_duration

### DISCUSSION THREAD

**Discussion by vekteo** [jspsych/jsPsych#743] [Q&A] [answered]
Hi all,

I'm trying to make an experiment with the SRT plugin. I want to set 120 ms as pre_target_duration, and that the button press ends the trial.

If I understand right, the "rt" will show the difference between the appearance of the stimulus and the corresponding button press.

However, as there is a pre_target_duration of 120 ms, it is possible that the user reacts _earlier_ than the appearance of the stimulus in the trial. Will this button press appear in the output file or it won't be recorded by default?

Thank you very much,

Teodora

**Comment by jodeleeuw** [comment, marked as answer]
The plugin code doesn't start listening for a keyboard response until after the target appears, so any responses prior to that would be ignored. You can see in the [source code](https://github.com/jspsych/jsPsych/blob/master/plugins/jspsych-serial-reaction-time.js#L131-L149) that the keyboard listener is created after the target is displayed.

If you want to capture anticipatory responses then you could modify the plugin and move the keyboard listener code outside the function that displays the target. The RT value would then be relative to the onset of the entire stimulus. You could correct the RT after the fact, or correct it in the plugin by subtracting the `pre_target_duration` parameter from the RT prior to the end of the trial.

**Reply by jodeleeuw** [reply]
Oh, I should mention here that if you do the correction, it's unlikely that a value of 0 will precisely reflect the onset of the stimulus. It's impossible in JavaScript to get the response timer to measure relative to the exact moment the stimulus was displayed, so it'll be off by a few tens of milliseconds most likely.

**Reply by vekteo** [reply]
Thank you!
